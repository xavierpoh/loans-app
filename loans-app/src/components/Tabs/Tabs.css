.tabs-container {
    display: flex;
    justify-content: center;
    align-items: center;
}

.tab-btn {
  width: 50%;
  height: 46px;
  border: 1px solid #019789;
  display: flex;
  justify-content: center;
  align-items: center;
  color: #019789;
  text-transform: uppercase;
  letter-spacing: .5px;
  font-size: 12px;
  font-weight: 400;
  cursor: pointer;
}

.tab-btn.left-tab {
  border-radius: 2px 0 0 2px;
}

.tab-btn.right-tab {
  border-radius: 0 2px 2px 0;
}

.tab-btn:hover {
  background-color: rgba(43,174,128,.2);
  color: #019789;
}

.tab-btn.is-selected {
  color: #fff;
  background: #019789;
}