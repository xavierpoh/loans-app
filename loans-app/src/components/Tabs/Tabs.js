import React, { Component } from 'react';
import './Tabs.css';

class Tabs extends Component {

  handleClick(tabIndex) {
    this.props.onTabChange(tabIndex);
  }

  render() {
    return (
      <div className="tabs-container">
        {this.props.tabList.map((tab, index) => {
          return (
            <div 
              className={`tab-btn ${index === this.props.currentTab ? 'is-selected' : ''} ${index === 0 ? 'left-tab': 'right-tab'}`} 
              onClick={() => this.handleClick(index)} 
              key={index}>
              {tab}
            </div>
          )
        })}
      </div>
    )
  }
}

export default Tabs;