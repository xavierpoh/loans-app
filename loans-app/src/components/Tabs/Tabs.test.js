import React from 'react';
import { shallow, mount } from 'enzyme';
import Tabs from './Tabs';

const TabList = ['Loan Application', 'Loan Repayment'];

describe('Tabs', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Tabs tabList={TabList} />);
    expect(wrapper).toMatchSnapshot();
  })
});

test('TabsComponent calls handleClick when Tab is clicked', () => {
  const handleClick = jest.fn();
  const wrapper = mount(
    <Tabs tabList={TabList} onTabChange={handleClick} /> 
  );
  const rightTab = wrapper.find('.right-tab');
  rightTab.simulate('click');
  expect(handleClick).toBeCalledWith(1);
});