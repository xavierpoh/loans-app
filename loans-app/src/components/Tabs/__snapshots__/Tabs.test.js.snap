// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tabs
    tabList={
        Array [
            "Loan Application",
            "Loan Repayment",
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="tab-btn  left-tab"
          onClick={[Function]}
>
          Loan Application
</div>,
        <div
          className="tab-btn  right-tab"
          onClick={[Function]}
>
          Loan Repayment
</div>,
      ],
      "className": "tabs-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": "Loan Application",
          "className": "tab-btn  left-tab",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Loan Application",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": "Loan Repayment",
          "className": "tab-btn  right-tab",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Loan Repayment",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="tab-btn  left-tab"
            onClick={[Function]}
>
            Loan Application
</div>,
          <div
            className="tab-btn  right-tab"
            onClick={[Function]}
>
            Loan Repayment
</div>,
        ],
        "className": "tabs-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": "Loan Application",
            "className": "tab-btn  left-tab",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Loan Application",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": "Loan Repayment",
            "className": "tab-btn  right-tab",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Loan Repayment",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
